Firestore/core/src/firebase/firestore/remote/connectivity_monitor_noop.cc

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/remote/connectivity_monitor.h"
+
+#include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
+#include "absl/memory/memory.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+using util::AsyncQueue;
+
+// Returns the default monitor that does nothing. This is to ensure that
+// build doesn't break on platforms which don't yet implement
+// `ConnectivityMonitor`.
+std::unique_ptr<ConnectivityMonitor> ConnectivityMonitor::Create(
```

Comment:
Doesn't the no-op implementation need to send an initial available callback?

Reply:
Hmm, currently the calls only react to changes in state and don't use the initial state in any way. I'm not against adding this, but I'd prefer to do it once there's a use case.